core/common/src/main/java/alluxio/AbstractThriftClient.java

```diff
@@ -165,9 +169,10 @@
    *
    * @param client the client
    * @param e the exception
+   * @throws E if the exception is not suppressed
    */
-  protected void processException(C client, Exception e) {
-    // Empty implementation by default.
+  protected <E extends Exception> void processException(C client, E e) throws E {
```

Comment:
do we really need this method? it is only used in this abstract class but never being extended